x/posts/internal/keeper/handler.go

```diff
@@ -172,3 +178,112 @@ func handleMsgRemovePostReaction(ctx sdk.Context, keeper Keeper, msg types.MsgRe
 	}
 	return &result, nil
 }
+
+// checkPostPollValid perform checks to ensure the validity of a poll
+func checkPostPollValid(ctx sdk.Context, id types.PostID, keeper Keeper) (*types.Post, error) {
+	// checks if post exists
+	post, found := keeper.GetPost(ctx, id)
+	if !found {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("post with id %s doesn't exist", id))
+	}
+
+	// checks if post has a poll
+	if post.PollData == nil {
+		return &post, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("no poll associated with ID: %s", id))
+	}
+
+	// checks if the poll is already closed or not
+	if time.Now().UTC().After(post.PollData.EndDate) {
```

Comment:
As we have the `Open` field, I would suggest using that field to check whether the post is closed or not. This is due to the fact that time checks might have different results based on the node that is performing the check itself. 

Suppose we have a network that has two nodes, and poll that has closing time at 11:30:00. Now, this happens: 

* node A checks the end date at 11:29:59.999 (1 ms before it), and the check returns true 
* node B checks the end data at 11:30:00.001 (1 ms after it), and the check returns false 

In this case, node A and node B would have two incoherent states and will later most probably produce some bugs. 
Instead, if we use the `Open` field than this will not happen.